lib/Parser/parsing.cpp

```diff
@@ -29,12 +29,13 @@ const SourceFile *Parsing::Prescan(const std::string &path, Options options) {
     }
   }
 
-  std::stringstream fileError;
+  std::string fileError_buf;
+  llvm::raw_string_ostream fileError{fileError_buf};
```

Comment:
How is llvm::raw_string_ostream better than std::stringstream?

Reply:
Use of llvm::raw_ostream over std::ostream is [mandated for new code going in to llvm](https://llvm.org/docs/CodingStandards.html#use-raw-ostream) and this has been bought up on the mailing list with respect to f18, so if our goal is to submit to llvm need to change over.

Reply:
There is no good reason not to use std::stringstream and it's not clear that's what that link says. Right above there it says:
> Note that using the other stream headers (`<sstream>` for example) is not problematic in this regard

Reply:
It does say that sstream is not problematic in the regard that it doesn't introduce global objects with non-static constructors. However, it goes on to say:
> New code should always use raw_ostream for writing

Which I think is fairly clear, and we certainly are new code from LLVM's perspective.

Reply:
I believe the context of the quote "New code should always use...." pertains to writing files.  In this case, stringstream seems better and appropriate.

Reply:
I believe this was specifically clarified by Hal on a community call a few weeks ago, that the policy is that the use of any of the stream libraries including `std::stringstream` is highly discouraged for new code.

Could you explain why stringstream is better and more appropriate here? `std::stringstream` and `llvm::raw_string_ostream` have the same interface.

Reply:
> Could you explain why stringstream is better and more appropriate here?

It's part of the language.

> `std::stringstream` and `llvm::raw_string_ostream` have the same interface.

This change shows a case where that's not the case.

Reply:
> Could you explain why stringstream is better and more appropriate here?

And `llvm::raw_string_ostream` is a part of the library of the project we are supposed to be a part of, and is preferred by that project for a number of technical reasons that are outlined in the documentation. I fail to see how this makes `std::stringstream` better.

> This change shows a case where that's not the case.

Are you referring to the fact that a separate buffer needs to be stored here? That is a minor change in the construction of the class but the observable interface as used here is identical.

Reply:
We discussed this on the flang technical call on March 9 and agreed:
 -  The LLVM Coding Guidelines are unclear on whether or not sstream should be allowed
 -  The long-time LLVM community folks felt that the intention is for sstream not to be used at all in preference to LLVM APIs.
 - We would make the change that David suggests in F18
 - Johannes would start a thread to clarify the wording in the LLVM Coding Standards to make clear the intent.

Reply:
The use of `std::stringstream` in Polly was removed: https://github.com/llvm/llvm-project/commit/0e93f3b0a06fc470e37b5be99734f5105ba11232.